src/__test__/productList.test.js

```diff
@@ -0,0 +1,29 @@
+import { render } from '@testing-library/react';
+import React from 'react';
+import { Provider } from 'react-redux';
+import App from '../App';
+import store from '../redux/store';
+import { productReducer } from '../redux/action/productAction';
+
+describe('test App component', () => {
+  test('check whether the component renders correctly', () => {
+    const tree = render(
+      <React.StrictMode>
+        <Provider store={store}>
+          <App />
+        </Provider>
+      </React.StrictMode>,
+    );
+    expect(tree).toMatchSnapshot();
+  });
+});
+
+it('Expect Stocks reducer to return the initial state', () => {
+  expect(productReducer(undefined, {})).toEqual({
+    products: [],
+  });
+});
+
+it('Expect productReducer', () => {
+  expect(productReducer({}, {})).not.toBeNull();
+});
```

Comment:
- [ ] Following the project requirements and as requested by the previous reviewers, you should add unit tests for several components (ProductContent, ProductDetail ...) of the app and some pure functions in Redux code (ex: selectAddReducer).
- [ ] Following the project requirements, you should use also React Testing Library to simulate user interaction. Check out this [link ](https://github.com/microverseinc/curriculum-react-redux/blob/main/math-magicians/articles/testing_react_with_react_testing_library.md)to learn how to do that.